src/ocr/ocr.cpp

```diff
@@ -83,6 +83,12 @@ void Ocr::recognize(const QPixmap &pixmap, int dpi)
     m_future = QtConcurrent::run([this, dpi, image = pixmap.toImage()] {
         m_tesseract.SetImage(image.constBits(), image.width(), image.height(), image.depth() / 8, image.bytesPerLine());
         m_tesseract.SetSourceResolution(dpi);
+        QMap<QString, QVariant> parameters = AppSettings().tesseractParameters();
```

Comment:
Is it necessary to set variables on every run? Can we just set it once at loading settings in `MainWindow::loadAppSettings()`? This will require to add a setter that accepts `QMap<QString, QVariant>` in `Ocr` class, but will work much more efficient (this place is a bottleneck on low end computers).

Reply:
Which part exactly is slow, the call to `SetVariable` or reading the parameters from settings? Glancing through the source, it doesn't look like `SetVariable` does much besides check that the value is the right type for they key and store it in an internal structure.

`loadAppSettings()` only gets called at the beginning and when the user closes the settings dialog. I'm editing the parameters in the config file directly with a keyboard shortcut bound to a script because I don't want to toggle five or six parameters in the UI every time I need to switch between horizontal and vertical text. The way it's currently written picks up changes from the config file too.

That being said, the parameters certainly don't need to be set every time `recognize()` is called. Outside of languages written in multiple directions, I doubt most people will be editing them regularly. We could store a copy of the parameters in a private field in `Ocr` and only call `SetVariable` if there is a change.

Reply:
Reading the settings from file itself is not very fast... I just have friends who complain about this function :) But I got your point. Maybe export OCR variables to D-Bus? Should be easy to use and fast.

Reply:
Ah that makes sense. D-Bus sounds like a good solution to me.

Reply:
Nice! I will export it and ping you in a separate PR in a few days. For now, let's leave it as it is.